### ISSUE TITLE
support for application/x-www-form-urlencoded

### ISSUE THREAD

**Issue by szaher** [deepmap/oapi-codegen#523]
Does oapi-codegen support content type like application/x-www-form-urlencoded ?

**Comment by mromaszewicz** [comment]
Yes, application/x-www-form-urlencoded is fully supported. When a request body uses this content type, oapi-codegen generates form-tagged structs (via the Formdata tag in operations.go). The client generates methods like OperationWithFormdataBody() using runtime.MarshalForm(), and the strict server parses form data via r.ParseForm() and runtime.BindForm(). There are dedicated test cases in internal/test/strict-server/ demonstrating full end-to-end support. Closing as this has been implemented.
